src/java.base/share/classes/java/nio/MappedByteBuffer.java

```diff
@@ -116,28 +116,38 @@ public abstract sealed class MappedByteBuffer
     }
 
     UnmapperProxy unmapper() {
-        return fd != null ?
-                new UnmapperProxy() {
-                    @Override
-                    public long address() {
-                        return address;
```

Comment:
We can just declare a `final long address = MappedByteBuffer.this.address;` instead of a huge refactor, right?